src/history/rdf/graph.rs

```diff
@@ -0,0 +1,217 @@
+#![allow(
+    clippy::module_name_repetitions,
+    clippy::min_ident_chars,
+    clippy::pattern_type_mismatch
+)]
+/// The helper methods for working with RDF in Stelae.
+use anyhow::Context;
+use sophia::api::graph::{GTripleSource, Graph};
+use sophia::api::ns::NsTerm;
+use sophia::api::MownStr;
+use sophia::api::{prelude::*, term::SimpleTerm};
+use sophia::inmem::graph::FastGraph;
+use std::iter;
+/// Stelae representation of an RDF graph.
+pub struct StelaeGraph {
+    /// The underlying graph.
+    pub g: FastGraph,
+}
+
+impl Default for StelaeGraph {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl StelaeGraph {
+    /// Create a new graph.
+    #[must_use]
+    pub fn new() -> Self {
+        Self {
+            g: FastGraph::new(),
+        }
+    }
+    /// Extract a literal from a triple matching.
+    ///
+    /// # Errors
+    /// Errors if the triple matching the object is not found.
+    /// Errors if the object is not an RDF literal.
+    pub fn literal_from_triple_matching(
+        &self,
+        subject: Option<&SimpleTerm>,
```

Comment:
That suggestion sounds reasonable, but I believe it would be more readable right now without struct fields.  We pass in these args differently when inserting history, so in that case we'd need to reinitialize the struct fields often.